src/main/java/fi/hsl/pulsarmqttsource/Main.java

```diff
@@ -1,13 +1,81 @@
 package fi.hsl.pulsarmqttsource;
 
-/**
- * Hello world!
- *
- */
-public class Main
-{
-    public static void main( String[] args )
-    {
-        System.out.println( "Hello World!" );
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigFactory;
+import java.io.File;
+
+/** Main reads the configuration, starts logging and passes the baton onwards. */
+public class Main {
+
+  public static void main(String[] args) {
+    Config config = createConfig(args);
+    System.out.println(config);
+    // Foo.run(conf);
+  }
+
+  /**
+   * Create a valid Config from a configuration file and environment variables.
+   *
+   * <p>If the environment variable CONFIG_PATH is set, it determines the path to the configuration
+   * file. Otherwise only the other environment variables affect the configuration.
+   *
+   * <p>Environment variables override values in the configuration file in case of conflict.
+   *
+   * @param args Command-line arguments. Currently ignored.
+   * @return Complete and valid configuration.
+   */
+  private static Config createConfig(@SuppressWarnings("unused") String[] args) {
```

Comment:
Looks good! We could name the class to be HslApplication or similar and then create all our apps by just extending this class. If we do so, let's change these private methods to protected and possibly omit the static also so we can override the methods in subclasses if needed.